### DISCUSSION TITLE
Trigger fault expansion card

### DISCUSSION THREAD

**Discussion by secumundo** [Dilbert66/esphome-vistaECP#76] [General]
Hello .. first an foremost: Superb work. Thank you. I modified your example and threw all the MQTT stuff out because I want to hook up to the system differently. Now that everything is working (I forgot to configure the ESP8266 as a second terminal) I have new ideas how to (ab)use the module almost every minute. I want to use my outdoor cams (with deepstack person recognition) to be able to trigger an alert. I understand that the module supports virtual 'expanded' zones and  configured "#define ZONEEXPANDER1 11" accordingly which should give me zone 41. However vista.setExpFault(41,false) or vista.setExpFault(41,true) doesn't do anything .. so I guess I am missing something.

Any help is appreciated.

Michaela

**Comment by Dilbert66** [comment]
You also need to program the expanded zones in program *56

**Reply by secumundo** [reply]
D'ohh .. of course. Thanks. Should I program the Zone as NO or NC?

**Comment by Dilbert66** [comment]
I use NC

**Reply by secumundo** [reply]
Super. Thanks.

**Reply by secumundo** [reply]
Well .. bad luck. It seems my Vista-15 doesn't allow anything > zone 24 .. and zones 9-24 are occupied by a wireless receiver. Any other idea how to trigger an alarm from the ESP ? Short of using a relay?

**Reply by Dilbert66** [reply]
ah.. I can't think of anything at this point unless you want to trigger a panic alarm or something using keypad commands sent from the ESP.
The other option is buy a used vista 20p panel board.  They are fairly inexpensive on ebay.  Your peripherals will work with it.

**Reply by secumundo** [reply]
AFAIK your program isn't able to send two key presses at the same time currently .. ?

**Reply by Dilbert66** [reply]
No but there is no need as the combined keypresses is only used by the keypad itself to determine what code to send to the panel.  You can send any cmd to the panel.

**Reply by secumundo** [reply]
What Code is default for panic? Didn't find anything in the source .. ?

**Reply by Dilbert66** [reply]
it depends on which function key it's programmed on A, B,C,D.  You need to program it first. Then you just write the correspending letter A,B,C or D

**Reply by Dilbert66** [reply]
Here is what gets sent when *1 (zone 95 alarm - fire):    0x0c
Zone 99 Panic: *#  sends 0X0D 
zone 96 alarm #3 sends 0x0E

**Reply by Dilbert66** [reply]
If you want to see what gets sent from the keypad, just look at the F6 EXT command. It will contain the actual keys sent to the panel.

**Reply by Dilbert66** [reply]
Actually I noticed that I did not provide the functionality to trigger panic zones 95,96 and 99 directly. I've pushed an update to dev that adds key commands "F", "P" and "M" to trigger corresponding panic zones 95,96 and 99
